### ISSUE TITLE
Add banners to the VBA Facility content type

### ISSUE THREAD

**Issue by xiongjaneg** [department-of-veterans-affairs/va.gov-cms#14700]
## Description
Drupal implementation for banner content type is built to match the spec #13811 
- Refer to VBA Content Specs XLS linked in #13096

## Acceptance Criteria
- [x] Replicate the full-width alert for these fields
- [x] Conditionally enabled banner doobley-doo (expandable fieldset) from CLP model
  - [x] Fields are cleared out when Use the banner is disabled
- [x] Fields appear within fieldsets and in order as shown in the spec
  - [x] Fields for alert-title and alert-body
  - [x] Alert type field (Information or Warning)
- [x] Help text is provided by UX designer
- [x] CLP expandable fieldset updates are tested (see CLP test as the model)
- [x] Add link in accessibility audit ticket #14896 for Laura on where to review implementation
- [x] Follow up tickets are created to track any remaining work that arises during implementation

**Comment by swirtSJW** [comment]
Plans documented here https://github.com/department-of-veterans-affairs/va.gov-cms/issues/13811

**Comment by xiongjaneg** [comment]
@xiongjaneg final decision on options

**Comment by xiongjaneg** [comment]
Assuming the three fields from option 4, does this need a design ticket? What pattern might be used? Something similar to campaign landing page where it's hidden until the box is checked? Or that it's visible in a field group?

RE is a design ticket needed? This is relatively small but there is a dependency for front end.

**Comment by xiongjaneg** [comment]
@davidmpickett @swirtSJW @omahane Please comment with any additional information or questions to help further define and size this ticket. I'll add as a candidate for sprint 95 with the understanding that it may need some clarification in sprint planning. Thank you! cc @jilladams

**Comment by xiongjaneg** [comment]
@swirtSJW @omahane Can this one be refined and pointed async now that the [spec ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/13811) is done? Wondering if we can get it refined in time for next sprint.

**Comment by omahane** [comment]
i would give this a 5.

**Comment by davidmpickett** [comment]
> @thejordanwood FYI that during Drupal prefinement for Facilities today, we discussed having you help out on this ticket as part of ACs vs. creating a separate ticket, and want to confirm if that makes sense for you or you'd prefer a separate ticket. The work for you will be Help text on some of the fields that will be added.
> 
> And making the timing note: if we opt for a separate ticket, it'll need to be planned into the same sprint as this ticket, to work in parallel.
> 
> cc @xiongjaneg

FYI - We do already have a ticket for UX review of banners #14844. I have previously suggested that ticket is not necessary because we also have a ticket for reviewing the content type they are being added to #14888 (also ditto the corresponding accessibility tickets). Having them as separate tickets is partially an artifact of a previous time when banners were planned as a separate content type

**Comment by thejordanwood** [comment]
@jilladams @xiongjaneg @davidmpickett 

1. I can help out on this ticket. Are we adding 3 fields – Fields for alert-title, alert-body, and alert type fields (Information or Warning)? Will this look similar to the [full width alert content type](https://staging.cms.va.gov/node/add/banner)? 
2. I agree with Dave. Let's close #14844 since we're okay with doing the UX review within this ticket.

**Comment by xiongjaneg** [comment]
Closed #14844

**Comment by davidmpickett** [comment]
**MUST**

- field_banner_content should use the widget `Text area (multiple rows) with counter` and be limited to 300 characters. This matches the limitation on VAMC banners and [Design System guidance](https://design.va.gov/components/banner/#content-considerations).

**SHOULD**
- Change "Persistence" to "Dismissible by user?" and make "Allow site visitors to dismiss banner ~~permanently~~" the first choice and selected by default to match [Design system guidance](https://design.va.gov/components/banner/#how-to-use-banner)
- "Enable this page segment" -> "Display a banner alert on this facility"

CONSIDER
- Most of these are really questions/ideas for @thejordanwood / @ALogsdon3 to consider
- Should all instances of "banner" be "banner alert"? It is both a banner and an alert and fully understanding it requires referencing both the Banner and Alert Design System components.
- Update the field group description (This optional segment allows you to add a banner to the VBA facility.) to have more explicit guidance about the limited use cases for adding a banner alert (e.g. actual emergencies) and potentially link to this KB article (https://prod.cms.va.gov/help/vamc/about-locations-content-for-vamcs/about-alerts-and-operating-statuses/best-practices-for-vamc-system-banner-alerts)
- "Banner title" -> "Banner headline" to match terminology in [Design System](https://design.va.gov/components/banner/)
- "Banner content" -> " "Banner message" to match terminology in [Design System](https://design.va.gov/components/banner/)
- Consider something like this help text for Banner type field (copied from VAMC banner alert)

> <p>Currently, the CMS supports two types of alerts:</p>
> <ol>
> 	<li><strong>Informational alerts</strong>, which are used to provide helpful information to a user or something that warrants attention. Not used for negative consequences.</li>
> 	<li><strong>Warning alerts</strong>, which are used to warn a user, such as when there are negative consequences, but necessary when something has gone wrong.</li>
> </ol>
> <p>You can read more about these alert types in the <a href="https://design.va.gov/components/alertboxes">VA Design System</a>.</p>

**Comment by omahane** [comment]
##Setting required fields

<img width="869" alt="Screenshot 2023-11-14 at 8 47 32 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/766573/efca98b9-c438-497a-85f4-ceae5f72ab6c">


### Banner Type
Banner Type appears to be required, having a red asterisk, but it is not required. You can submit the form without it.

#### Possible solution
Use constraint validator.

### Dismissable
- Dismissable doesn't have a red asterisk.
- Is not required to be filled out.
- Has N/A
- Default value not set

#### Possible solution
##### Make dismissable required
- removes the N/A
- Set the value at "Allow"

Do a bulk update after deploy
Set the value during migration.
##### Address each issue
- Use alter statement to remove N/A
- Use JS to apply required classes
- Use constraint validator to make sure it's been filled out

**Comment by thejordanwood** [comment]
I added a _VBA Banners_ page to [the Figma file](https://www.figma.com/file/aKdT4weNNthDxb3r6MolEh/VBA-Content-Specs?type=design&node-id=362%3A770&mode=design&t=xWfG0zwPvlFANqqu-1) to document the help text changes that are being suggested.

**Comment by swirtSJW** [comment]
I just realized we neglected to include test coverage on this PR.  So I created [a followup ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16146).

**Comment by davidmpickett** [comment]
Uploading video for async demo:
https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/0b60499d-20c8-410d-a8c2-4fbb605f1bce
